packages/lib-sourcify/src/lib/solidityCompiler.ts

```diff
@@ -140,6 +170,51 @@ async function fetchSolcFromGitHub(
   return false;
 }
 
+async function validateSolcJsPath(soljsonPath: string): Promise<boolean> {
+  try {
+    const solcjsImports = await import(soljsonPath);
+    const soljson = solc.setupMethods(solcjsImports);
+    if (soljson) {
+      return true;
+    }
+    return false;
+  } catch (e) {
+    return false;
+  }
+}
+
+async function fetchSolcJsFromGitHub(
+  solcJsPath: string,
+  version: string,
+  fileName: string
+): Promise<boolean> {
+  const githubSolcURI = GITHUB_SOLCJS_REPO + encodeURIComponent(fileName);
+  const res = await fetchWithTimeout(githubSolcURI);
+  // TODO: Handle nodejs only dependencies
+  if (res.status === StatusCodes.OK) {
+    fs.mkdirSync(path.dirname(solcJsPath), { recursive: true });
+    const buffer = await res.arrayBuffer();
+
+    try {
+      fs.unlinkSync(solcJsPath);
+    } catch (_e) {
+      undefined;
+    }
```

Comment:
Any ideas why we `unlink()`? I see this was also in the old `fetcSolc...` code

Reply:
Honestly I didn't check how this work, I'll spend some time understanding how this function works.

Reply:
Maybe because it fails to write if the file is already there?